### PR TITLE
Channels: Add sort options to streams

### PR DESCRIPTION
**Pull request by src-tinkerer**
Uses the same method that is used for videos. We create continuation based on the sort option. One note though, `produce_channel_livestreams_continuation` and `produce_channel_videos_continuation` are almost identical, so is it better to simply make a singular function that produces the continuation based on the content?

### CODE REVIEW

**Comment by SamantazFox** [comment]
Yes, merging the two in a single function is strongly preferred.
Also, please remove `produce_channel_livestream_url`. It has no use (the function you copied, `produce_channel_videos_url` was only used to bypass captchas back then and will be discarded one day).

**Comment by src-tinkerer** [comment]
I made the requested changes but couldn't test it yet because YouTube is blocked in my country, shouldn't be merged until then.

**Review by SamantazFox — CHANGES_REQUESTED** [review]
Code looks good! I'm only asking for a few style/syntax changes ^^

File: src/invidious/channels/videos.cr [hunk 1]
```diff
@@ -1,4 +1,4 @@
-def produce_channel_videos_continuation(ucid, page = 1, auto_generated = nil, sort_by = "newest", v2 = false)
+def produce_channel_content_continuation(ucid, content, page = 1, auto_generated = nil, sort_by = "newest", v2 = false)
```
[the remark this answers is not in the document]
Comment (src-tinkerer) [reply src/invidious/channels/videos.cr]: Done

File: src/invidious/routes/channels.cr [hunk 2]
```diff
@@ -81,13 +81,12 @@ module Invidious::Routes::Channels
       return env.redirect "/channel/#{channel.ucid}"
     end
 
-    # TODO: support sort option for livestreams
-    sort_by = ""
-    sort_options = [] of String
+    sort_by = env.params.query["sort_by"]?.try &.downcase
```
[the remark this answers is not in the document]
Comment (src-tinkerer) [reply src/invidious/routes/channels.cr:84]: > For consistency with what you did in `routes/api/v1/channels.cr` (don't forget to also change the line underneath, as I can't make a suggection across two separate diff chunks)

The thing is this is how it's done with videos in `src/invidious/routes/channels.cr`. Also what line underneath? `sort_options = {"newest", "oldest", "popular"}` ?

File: src/invidious/channels/videos.cr (hunk 1 above)
[the remark this answers is not in the document]
Comment (src-tinkerer) [reply src/invidious/channels/videos.cr]: Done

File: src/invidious/channels/videos.cr [hunk 3]
```diff
@@ -144,21 +151,20 @@ module Invidious::Channel::Tabs
...
   # -------------------
 
-  def get_livestreams(channel : AboutChannel, continuation : String? = nil)
+  def get_livestreams(channel : AboutChannel, continuation : String? = nil, sort_by = "newest")
     if continuation.nil?
-      # EgdzdHJlYW1z8gYECgJ6AA%3D%3D is the protobuf object to load "streams"
-      initial_data = YoutubeAPI.browse(channel.ucid, params: "EgdzdHJlYW1z8gYECgJ6AA%3D%3D")
-    else
-      initial_data = YoutubeAPI.browse(continuation: continuation)
+      continuation ||= make_initial_content_ctoken(channel.ucid, "livestreams", sort_by)
```
[the remark this answers is not in the document]
Comment (src-tinkerer) [reply src/invidious/channels/videos.cr]: Done

File: src/invidious/routes/channels.cr (hunk 2 above)
[the remark this answers is not in the document]
Comment (SamantazFox) [reply src/invidious/routes/channels.cr:84]: In the line where you're doing `(sort_by || "newest")` !

File: src/invidious/routes/channels.cr [hunk 4]
```diff
@@ -81,13 +81,12 @@ module Invidious::Routes::Channels
...
+    sort_options = {"newest", "oldest", "popular"}
 
     # Fetch items and continuation token
     items, next_continuation = Channel::Tabs.get_60_livestreams(
-      channel, continuation: continuation
+      channel, continuation: continuation, sort_by: (sort_by || "newest")
```
Comment (SamantazFox) [inline src/invidious/routes/channels.cr:89]: To go with my other code suggestion:
```suggestion
      channel, continuation: continuation, sort_by: sort_by
```

**Comment by src-tinkerer** [comment]
Done, sorry for delay. Been busy with midterms.

**Review by SamantazFox — APPROVED** [review]

**Comment by src-tinkerer** [comment]
Will test this week

**Comment by src-tinkerer** [comment]
Okay done testing and live stream sort works.

**Review by SamantazFox — APPROVED** [review]

**Comment by src-tinkerer** [comment]
Is this getting merged? xD

**Comment by SamantazFox** [comment]
Thanks for your contribution :)

**Pull request merged**